presto-main-base/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -930,19 +931,27 @@ public PlanOptimizers(
         // To replace the JoinNode to MergeJoin ahead of AddLocalExchange to avoid adding extra local exchange
         builder.add(new MergeJoinForSortedInputOptimizer(metadata, featuresConfig.isNativeExecutionEnabled()));
 
+        builder.add(
+                new IterativeOptimizer(
+                        metadata,
+                        ruleStats,
+                        statsCalculator,
+                        costCalculator,
+                        new ExtractSystemTableFilterRuleSet(metadata.getFunctionAndTypeManager()).rules()));
```

Comment:
Should we move this to L992 above the `// DO NOT add optimizers that change the plan shape (computations) after this point` comment